api/app/db/crud/weather_models.py

```diff
@@ -241,6 +241,82 @@ def get_station_model_predictions(
     return query
 
 
+def get_latest_station_model_prediction_per_day(session: Session,
+                                                station_codes: List,
+                                                model: str,
+                                                start_date: datetime.datetime,
+                                                end_date: datetime.datetime):
+    result = session.execute(f"""
+                    SELECT DISTINCT ON (unique_day)
+                    weather_station_model_predictions.id,
+                    max(weather_station_model_predictions.prediction_timestamp) AS latest_prediction_timestamp,
+                    weather_station_model_predictions.station_code,
+                    weather_station_model_predictions.rh_tgl_2,
+                    weather_station_model_predictions.tmp_tgl_2,
+                    weather_station_model_predictions.bias_adjusted_temperature,
+                    weather_station_model_predictions.bias_adjusted_rh,
+                    weather_station_model_predictions.delta_precip,
+                    weather_station_model_predictions.wdir_tgl_10,
+                    weather_station_model_predictions.wind_tgl_10,
+                    max(latest_timestamp) AS latest_run_timestamp,
+                    date(latest_timestamp) AS unique_day
+                FROM
+                    weather_station_model_predictions
+                    JOIN (
+                        SELECT
+                            prediction_model_run_timestamps.id AS latest_timestamp_id,
+                            max(prediction_run_timestamp) AS latest_timestamp,
+                            date(prediction_run_timestamp) AS day,
+                            weather_station_model_predictions.station_code
+                        FROM
+                            prediction_model_run_timestamps,
+                            weather_station_model_predictions
+                        WHERE
+                            prediction_model_id IN (SELECT id FROM prediction_models WHERE abbreviation = '{model}')
+                            AND station_code IN ({",".join(str(s) for s in station_codes)})
```

Comment:
I'd prefer to see this translated into Python/sqlalchemy. That way we automatically have protection against the most common SQL injection attack methods.

Reply:
We already implicitly have this since these parameters are validated by pydantic

Reply:
pydantic doesn't know any more than to expect `station_codes` to be a list. 
`station_codes = ['1); DROP table;']` will satisfy pydantic, and will also cause trouble.

Reply:
Not true, it knows it should be a list of integers so that query is not possible.

Reply:
where is the list of ints specified? I only see a generic List in the function signature

Reply:
This is what I get back when I try that request body:

```
{
  "detail": [
    {
      "loc": [
        "body",
        "stations",
        0
      ],
      "msg": "value is not a valid integer",
      "type": "type_error.integer"
    }
  ]
}
```

Reply:
> where is the list of ints specified? I only see a generic List in the function signature
In the schema definitions:

```
class ModelDataRequest(BaseModel):
    """ A request for weather data for a given set of stations with a time of interest. """
    stations: List[int]
```

I can adjust the typings of the functions to use the specified type too for clarity